#summary an example representing how to use lncRScan

= Introduction =

This is an example representing how to use lncRScan to detect novel lncRNAs

= Input =
high-confident.gtf -- a GTF file recording high-confident transcripts with classcodes

lnc_class -- a list of classcodes, which may contain lncRNAs

= Steps =

 * extract_category -- extract 3121 transcripts from input GTF file according to the classcodes listed.

   (1) $ extract_GTF_class.pl high-confident.gtf lnc_class (extracted.gtf and extracted_id are generated)

   (2) $ mv extracted.gtf transcripts_1.gtf

 * extract_length -- extract transcripts from transcripts_1.gtf according to length >=200nt

   (1) Get transcripts_1.bed according to transcripts_1.gtf using UCSC table browser

   (2) extract 3094 multi-exon transcript with length >=200nt
       $ extract_long_multi.pl transcripts_1.bed > transcripts_2.bed

   (3) get transcripts_2.fa according to transcripts_2.bed

 * extract_ORF -- get IDs of transcripts with ORF length < 300

   (1) get longest ORF length of transcripts_2.fa
       $ longorf.pl --input transcripts_2.fa > transcripts_2.orf.list

   (2) get 734 transcript ids with length < 300
       $ awk '($2<300){print $1}' transcripts_2.orf.list > transcripts_3.list

   (3) trim the IDs in transcripts_3.list
       $ perl -ne '@aa = split("\_");print $aa\[4\] , "_", $aa\[5\]' transcripts_3.list > transcripts_3.renamed.list

   (4) extract transcripts from transcripts_1.bed according to transcripts_3.renamed.list
       $ extract_BED_id.pl transcripts_3.renamed.list transcripts_2.bed > transcripts_3.bed

 * Run PhyloCSF (score<0 or fail)

   (1) Stitch 29 mammalian genome alignment according to transcripts_3.bed using "Stitch MAF blocks"

   (2) Download the 29 mammalian alignments and the BED file from Galaxy

   (3) Modify the species' name in the alignment FASTA file using 29mammals.sh
       $ 29mammals.sh 29_mammals_aln.fasta > 29_mammals_aln_renamed.fasta

   (4) Create a table connecting transcript_id and its coordinates using
       $ awk '{print $4"\t"$1"("$6"):"$2"-"$3}' transcripts_3.bed > id-co.table

   (5) Separate the file into files, each of which only contains one alignment
       $ mkdir file_pieces
       $ pre_phylocsf.pl id-co.table 29_mammals_aln_renamed.fasta