#summary Brief introduction of lncRScan

= Introduction =

lncRScan is part of our method for predicting functional lncRNAs from RNA-Seq data. It consists of two stages. One is to filter possible artificial transcripts according to local FPKM scores and number spliced reads while the other stage is to generate a list of high-confident lncRNAs according to two criteria, namely exonic length of transcript and longest ORF length.


= Tools used =

Samtools--
A tool for processing SAM/BAM files.

= Input files =

1) combined.gtf--
  A GTF file recording gene annotations assembled by ab initio transcriptome reconstruction like RABT or cufflinks. It is generated by comparing the predicted transcripts to the reference annotations.

2) all.bam --
  A BAM file combining all mapped reads for KLF1 KO and WT replicates. It is output by 'Initial assembly'.

3) reads.number --
  Record how many mapped reads in the all.bam

  "samtools view -c all.bam > reads.number"

	
4) assembly-classcode --
  A table depicting transcript IDs and their class codes assigned by cuffcompare

  "awk '{print $1\t$4;}' cuffcmp.tracking > assembly-classcode"

  where cuffcmp.tracking is an output file from cuffcompare, which was run in 'High-confident lncRNAs generation'.

5) cuffcmp.loci --
A file output by cuffcompare

6) cuffcmp.assemblies.gtf.tmap --
A file output by cuffcompare

7) non-overlap_list --

8) candidate_list --


= Method overview =